R/provider-openai.R

```diff
@@ -79,7 +74,7 @@ chat_openai <- function(
     extra_args = api_args,
     api_key = api_key
   )
-  Chat$new(provider = provider, turns = turns, echo = echo)
+  Chat$new(provider = provider, system_prompt = system_prompt, echo = echo)
```

Comment:
Oh, looks like this change has the unintended consequence of causing `system_prompt` to require a scalar string:

``` r
chat_openai(c("one", "two"))
#> Using model = "gpt-4o".
#> Error in `self$set_system_prompt()` at ellmer/R/chat.R:37:9:
#> ! `value` must be a single string or `NULL`, not a character vector.
```

Reply:
Maybe the system prompt logic in `normalize_turns()` should move into `Chat$new()`? https://github.com/tidyverse/ellmer/blob/97c07c826006b6c22c068805648d92737cbe5b13/R/turns.R#L137-L140

Reply:
Ooh yeah, and I should add an explicit test for that.